components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApplicationDAOImpl.java

```diff
@@ -416,9 +390,51 @@ public void addApplicationKeys(String appId, String keyType, OAuthApplicationInf
         }
     }
 
-    /**
-     * @see ApplicationDAO#getLastUpdatedTimeOfApplication(String)
-     */
+    public List<OAuthApplicationInfo> getApplicationKeys(String appId) throws APIMgtDAOException {
+        final String getApplicationKeysQuery = "SELECT CLIENT_ID, KEY_TYPE FROM AM_APP_KEY_MAPPING " +
+                "WHERE APPLICATION_ID = ?";
+        List<OAuthApplicationInfo> keyList = new ArrayList<>();
```

Comment:
Thank you for the information. We do have key type in primary key for app key mapping.

https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.core.feature/resources/dbscripts/mysql.sql#L266